### PR TITLE
Added tooltip for Refund Requested

### PR DESCRIPTION
**Pull request by andrepestana-aot**
*Issue #:* /bcgov/entity#8563

*Description of changes:*

- Added tooltip for Refund Request

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

File: src/components/existing-request/existing-request-display.vue [hunk 1]
```diff
@@ -749,4 +901,36 @@ export default class ExistingRequestDisplay extends Mixins(
...
+.v-tooltip__content:after {
+  content: "" !important;
+  position: absolute !important;
+  top: 100% !important;
+  right: 50% !important;
+  border-right: 10px solid transparent !important;
+  border-left: 10px solid transparent !important;
+  border-top: 8px solid RGB(73, 80, 87) !important;
+  transform: translateY(50%);
+}
```
Comment (cameron-freshworks) [inline src/components/existing-request/existing-request-display.vue:920]: I believe the tool tips were styled in the overrides already. Does this design call for something different than we've already done?

Check it out though, there is tool tips for top, bottom and right i think,

File: src/enums/payment-method.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export enum PaymentMethod {
+    DIRECT_PAY = 'DIRECT_PAY', // Credit Card
+    INTERNAL = 'INTERNAL', // Staff payment: Routing Slip and No Fee
+    DRAWDOWN = 'DRAWDOWN', // BCOL
```
Comment (cameron-freshworks) [inline src/enums/payment-method.ts:4]: Should this be `DRAW_DOWN` maybe? Feels like 2 words but i could be wrong :)

File: src/components/existing-request/existing-request-display.vue [hunk 3]
```diff
@@ -228,6 +260,14 @@ export default class ExistingRequestDisplay extends Mixins(
   /** The pending payment, if any. See mounted(). */
   private pendingPayment = null
 
+  /** Params used to format refund message (tooltip and modal). */
+  private refundParams = {
```
Comment (cameron-freshworks) [inline src/components/existing-request/existing-request-display.vue:264]: This object could probably benefit from an interface.

File: src/components/existing-request/existing-request-display.vue [hunk 4]
```diff
@@ -734,6 +775,117 @@ export default class ExistingRequestDisplay extends Mixins(
       this.$root.$emit('showSpinner', false)
     }
   }
+
+  private get isThereMoreThanOnePaymentMethod () {
```
Comment (cameron-freshworks) [inline src/components/existing-request/existing-request-display.vue:779]: This method and the ones below could probably benefit from return types and methods comments/doc block comments.

File: src/components/existing-request/existing-request-display.vue [hunk 5]
```diff
@@ -734,6 +775,117 @@ export default class ExistingRequestDisplay extends Mixins(
...
+              'Your Name Request has been cancelled and a refund request is being processed.<br/><br/>' +
+              'A credit will be applied to your BC Registries account.<br/>There may be a one day delay before the ' +
+              'credit will show on your transactions / statetements.'
+            this.refundParams.showStaffContact = false
+          } else {
+            // May happen when a PAD is not processed yet.
+            // It usually takes a day to be processed.
+            this.refundParams.refundLabel = 'Refund Not Processed'
+            this.refundParams.refundMessage =
+              'Your Name Request has been cancelled.<br/><br/>' +
```
Comment (cameron-freshworks) [inline src/components/existing-request/existing-request-display.vue:830]: Why the double <br>?

File: src/components/existing-request/existing-request-display.vue [hunk 6]
```diff
@@ -44,8 +44,29 @@
                 <v-col cols="12" class="request-status">
                   <span>Request Status:</span>
                   &nbsp;
-                  <span :class="isNotPaid ? 'app-red' : isPaymentProcessing ? 'app-green' : ''">
-                    {{ requestStatusText }}</span>
+                  <span
+                    :class="isRefundRequested ? '' :
+                            isNotPaid ? 'app-red' :
+                            isPaymentProcessing ? 'app-green' : ''">
```
Comment (cameron-freshworks) [inline src/components/existing-request/existing-request-display.vue:50]: This chained ternary is a little hard to follow. 
Have you looked into class binding the colors conditionally?

File: src/enums/payment-method.ts (hunk 2 above)
Comment (andrepestana-aot) [reply src/enums/payment-method.ts:4]: Thanks for checking but it is DRAWDOWN. At least it is the way it comes from pay-api.

File: src/components/existing-request/existing-request-display.vue (hunk 5 above)
Comment (andrepestana-aot) [reply src/components/existing-request/existing-request-display.vue:830]: image
 There's an empty line.

File: src/components/existing-request/existing-request-display.vue (hunk 6 above)
Comment (andrepestana-aot) [reply src/components/existing-request/existing-request-display.vue:50]: yes, actually I changed isPaymentProcessing logic and revert this part back the original. Wil come in the next commit.

File: src/components/existing-request/existing-request-display.vue (hunk 3 above)
Comment (andrepestana-aot) [reply src/components/existing-request/existing-request-display.vue:264]: 👍

File: src/components/existing-request/existing-request-display.vue (hunk 4 above)
Comment (andrepestana-aot) [reply src/components/existing-request/existing-request-display.vue:779]: 👍

File: src/components/existing-request/existing-request-display.vue (hunk 1 above)
Comment (andrepestana-aot) [reply src/components/existing-request/existing-request-display.vue:920]: They actually asked me to use a tooltip with 270px as it is the default but it doesn't look good:
image

the `:after` is to make the arrow be pointed down.

Comment (cameron-freshworks) [reply src/components/existing-request/existing-request-display.vue]: I understand the context, this style already exists (or close too) in `overrides.scss` 
You would need to add the content class to your tool tip, you might not need to duplicate the styling in the component itself. 

See class `top-tooltip` in overrides. Using this on the tool tip itself is by adding it as a content class.
`<v-tooltip top content-class="top-tooltip">`

If you need to override the width that can be set on the component itself as well.

File: src/components/existing-request/existing-request-display.vue (hunk 5 above)
Comment (cameron-freshworks) [reply src/components/existing-request/existing-request-display.vue:843]: Easy!

File: src/components/existing-request/existing-request-display.vue (hunk 6 above)
Comment (cameron-freshworks) [reply src/components/existing-request/existing-request-display.vue]: Sounds good!

File: src/components/existing-request/existing-request-display.vue (hunk 1 above)
Comment (andrepestana-aot) [reply src/components/existing-request/existing-request-display.vue]: ok. Got it. Thanks!

File: src/components/existing-request/existing-request-display.vue [hunk 7]
```diff
@@ -44,8 +44,31 @@
                 <v-col cols="12" class="request-status">
                   <span>Request Status:</span>
                   &nbsp;
-                  <span :class="isNotPaid ? 'app-red' : isPaymentProcessing ? 'app-green' : ''">
-                    {{ requestStatusText }}</span>
+                  <span
+                    :class="isNotPaid ? 'app-red' : isPaymentProcessing ? 'app-green' : ''">
```
Comment (severinbeauvais) [inline src/components/existing-request/existing-request-display.vue:48]: If the alterate class binding syntax doesn't make this any neater, consider using a getter in the script instead.

File: src/components/existing-request/existing-request-display.vue [hunk 8]
```diff
@@ -228,6 +262,14 @@ export default class ExistingRequestDisplay extends Mixins(
   /** The pending payment, if any. See mounted(). */
   private pendingPayment = null
 
+  /** Params used to format refund message (tooltip and modal). */
+  private refundParams: RefundParamsIF = {
+    'refundMessage': '',
+    'refundLabel': '',
+    'showStaffContact': false,
+    'showAlertIcon': false
```
Comment (severinbeauvais) [inline src/components/existing-request/existing-request-display.vue:270]: You don't need the quotes around the property names.

File: src/components/existing-request/existing-request-display.vue [hunk 9]
```diff
@@ -734,6 +778,136 @@ export default class ExistingRequestDisplay extends Mixins(
       this.$root.$emit('showSpinner', false)
     }
   }
+
+  /** Check if there is more than one payment method used in the payments. */
+  private get isThereMoreThanOnePaymentMethod (): boolean {
```
Comment (severinbeauvais) [inline src/components/existing-request/existing-request-display.vue:783]: This is enough new code (and specific enough) that I wonder if it should be its own (sub) component? I see this component is almost 1000 lines already!

**Review by severinbeauvais — APPROVED** [review]

**Review by cameron-freshworks — APPROVED** [review]

File: src/components/existing-request/existing-request-display.vue (hunk 1 above)
Comment (cameron-freshworks) [reply src/components/existing-request/existing-request-display.vue]: Awesome!

File: src/components/existing-request/existing-request-display.vue (hunk 7 above)
Comment (andrepestana-aot) [reply src/components/existing-request/existing-request-display.vue]: I created a getter to define what class to use here. Thanks.

File: src/components/existing-request/existing-request-display.vue (hunk 8 above)
Comment (andrepestana-aot) [reply src/components/existing-request/existing-request-display.vue]: 👍 done.

File: src/components/existing-request/existing-request-display.vue (hunk 9 above)
Comment (andrepestana-aot) [reply src/components/existing-request/existing-request-display.vue:789]: do you think I should move it to payment mixin?

Comment (severinbeauvais) [reply src/components/existing-request/existing-request-display.vue:789]: Without looking it deeply, this sounds like a good idea.

Does the payment mixin have access to `this.payments` (which is used in the methods)?

I say do it if it's easy and cleaner than this :)

Comment (andrepestana-aot) [reply src/components/existing-request/existing-request-display.vue]: ok. Moved some getters to payment-mixin.
